### ISSUE TITLE
changelog files should be rendered as Markdown on Hackage

### ISSUE THREAD

**Issue by ulysses4ever** [haskell/cabal#8857]
Otherwise, they look obnoxious.

Allegedly, you only need to rename changelog -> changelog.md

**Comment by Kleidukos** [comment]
Tangent, but what was the original reason for not uploading the changelog of cabal-install itself and instead refer to a link on GitHub?

**Comment by ulysses4ever** [comment]
Right… @Mikolaj would know better but if I was to guess, I'd say that otherwise it would be a pretty damn big file, which everyone would download only to read a bit at the top. It's not the extra bytes traveling over the network that are along but the sloppiness of the rendering engine: web browsers were not very good at rendering such cases. Unless you're using eww, of course.

**Comment by Mikolaj** [comment]
Artem may be right and this may be a good reason to keep it that way. But even oldest women don't know the original reasons.  My guess was that thanks to that indirection and to pointing to the master branch, you can easily fix the changelog after the package is on Hackage, which is then immutable.

**Comment by ulysses4ever** [comment]
I'm sorry, I just screwed up: I tried the VS Code in the browser thing ("codespace"?), renamed the file, and pushed some green button. That silly thing pushed right into master: no branch, no PR. It's crazy...

**Comment by ulysses4ever** [comment]
Of course, CI fails on it... Whitespace, specifically.

**Comment by ulysses4ever** [comment]
I reverted it in the same devilish style: by pushing right into `master`. Sorry, but this looked the most reasonable.

**Comment by Mikolaj** [comment]
No sweat. :)

**Comment by ulysses4ever** [comment]
This has been done.
